src/Tools/dotnet-monitor/HostBuilderHelper.cs

```diff
@@ -112,6 +138,23 @@ public static IHostBuilder CreateHostBuilder(HostBuilderSettings settings)
                 });
         }
 
+        private static void UpdatePropertiesWarningList(HostBuilderContext context, string warningMessage)
+        {
+            if (context.Properties.TryGetValue(HostBuilderResults.ResultKey, out object resultsObject))
+            {
+                if (resultsObject is HostBuilderResults hostBuilderResults)
+                {
+                    hostBuilderResults.Warnings.Add(warningMessage);
+                }
+            }
+            else
+            {
+                HostBuilderResults results = new HostBuilderResults();
```

Comment:
Could we add the HostBuilderResults instance to the properties collection upfront and then add warnings to it as necessary without having to continuously look up if the object exists?